src/ui/public/visualize/visualize.js

```diff
@@ -159,38 +153,30 @@ uiModules
             if ($scope.vis.type.requiresSearch && ['query', 'filters'].includes(keys[0])) {
               $scope.fetch();
             }
-            if (keys[0] === 'uiState') {
-            // uiState can be changed by other visualizations on dashboard. this makes sure this fires only if
-            // current visualizations uiState changed.
-              if (!oldUiState || oldUiState !== JSON.stringify($scope.uiState.toJSON())) {
-                oldUiState = JSON.stringify($scope.uiState.toJSON());
-                $scope.fetch();
-              }
-            }
           });
 
           $scope.$on('$destroy', () => {
             stateMonitor.destroy();
           });
-        } else {
-          const handleUiStateChange = () => { $scope.$broadcast('render'); };
-          $scope.uiState.on('change', handleUiStateChange);
-          $scope.$on('$destroy', () => {
-            $scope.uiState.off('change', handleUiStateChange);
-          });
         }
 
-        resizeChecker.on('resize',  () => {
```

Comment:
Remove the resize checker from `visualize`, since it's properly done in `visualization` now (see above).